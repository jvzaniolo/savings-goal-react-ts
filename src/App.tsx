import { Header } from './layout/Header';
import { SavingGoal } from './components/SavingGoal';
import { ReachDateProvider } from './contexts/ReachDateContext';

export function App(): JSX.Element {
  return (
    <div className="w-screen h-screen bg-background">
      <Header />
      <main className="flex justify-center">
        <ReachDateProvider>
          <SavingGoal />
        </ReachDateProvider>
      </main>
    </div>
  );
}
