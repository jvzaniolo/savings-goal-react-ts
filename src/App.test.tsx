import { App } from './App';
import { render } from '@testing-library/react';

describe('App', () => {
  it('should render the App component', () => {
    const component = render(<App />);

    expect(component).toMatchSnapshot();
  });
});
