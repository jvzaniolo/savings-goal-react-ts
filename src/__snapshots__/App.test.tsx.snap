// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render the App component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="w-screen h-screen overflow-auto bg-background"
      >
        <header
          class="py-6 px-14 bg-white"
        >
          <img
            alt="Origin"
            class="h-8"
            data-testid="logo"
            src="logo.svg"
          />
        </header>
        <main
          class="mt-8 sm:mt-12 flex flex-col items-center"
        >
          <div
            class="sm:max-w-[40rem]"
          >
            <h3
              class="mb-6 text-center text-lg sm:text-xl text-brand-primary"
            >
              Let's plan your
               
              <strong
                class="font-semibold"
              >
                saving goal.
              </strong>
            </h3>
            <div
              class="pt-8 p-10 rounded-lg shadow-level4 bg-neutral-white"
            >
              <form
                class="flex flex-col flex-wrap space-y-6"
              >
                <section
                  class="flex items-center space-x-4"
                >
                  <img
                    alt="Buy a house"
                    src="buy-a-house.svg"
                    width="64"
                  />
                  <div
                    class="space-y-1"
                  >
                    <h1
                      class="text-xl sm:text-2xl font-display font-medium text-blue-gray-800"
                    >
                      Buy a house
                    </h1>
                    <h2
                      class="text-sm sm:text-base text-blue-gray-400"
                    >
                      Saving Goal
                    </h2>
                  </div>
                </section>
                <div
                  class="flex flex-wrap gap-4"
                >
                  <div
                    class="flex flex-1 flex-col"
                  >
                    <label
                      class="mb-1 text-xs sm:text-sm text-blue-gray-800"
                      for="amount"
                    >
                      Total amount
                    </label>
                    <div
                      class="pl-1 p-3 group h-14 flex flex-nowrap items-center space-x-1 rounded border font-display font-medium text-xl sm:text-2xl border-blue-gray-50 text-blue-gray-600 transition-colors overflow-hidden focus-within:outline focus-within:outline-brand-secondary focus-within:outline-2 undefined"
                    >
                      <span
                        class="transition-colors text-blue-gray-100 group-focus-within:text-brand-secondary"
                      >
                        <svg
                          fill="currentColor"
                          height="24"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 16 16"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M4 10.781c.148 1.667 1.513 2.85 3.591 3.003V15h1.043v-1.216c2.27-.179 3.678-1.438 3.678-3.3 0-1.59-.947-2.51-2.956-3.028l-.722-.187V3.467c1.122.11 1.879.714 2.07 1.616h1.47c-.166-1.6-1.54-2.748-3.54-2.875V1H7.591v1.233c-1.939.23-3.27 1.472-3.27 3.156 0 1.454.966 2.483 2.661 2.917l.61.162v4.031c-1.149-.17-1.94-.8-2.131-1.718H4zm3.391-3.836c-1.043-.263-1.6-.825-1.6-1.616 0-.944.704-1.641 1.8-1.828v3.495l-.2-.05zm1.591 1.872c1.287.323 1.852.859 1.852 1.769 0 1.097-.826 1.828-2.2 1.939V8.73l.348.086z"
                          />
                        </svg>
                      </span>
                      <input
                        class="flex-1 focus:outline-none"
                        id="amount"
                        inputmode="decimal"
                        name="amount"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                  <div
                    class="flex flex-1 flex-col"
                  >
                    <label
                      for="reach-date"
                      tabindex="1"
                    >
                      <span
                        class="mb-1 text-xs sm:text-sm text-blue-gray-800"
                      >
                        Reach goal by
                      </span>
                      <div
                        class="h-14 py-1 px-2 flex rounded border items-center border-blue-gray-50  undefined"
                      >
                        <button
                          class="w-10 h-10 text-center rounded-full transition-colors focus:outline-none text-blue-gray-50 hover:bg-transparent cursor-not-allowed"
                          data-testid="reach-date-left-button"
                          tabindex="2"
                          type="button"
                        >
                          <svg
                            class="m-2"
                            fill="currentColor"
                            height="24"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 24 24"
                            width="24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M0 0h24v24H0V0z"
                              fill="none"
                            />
                            <path
                              d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                            />
                          </svg>
                        </button>
                        <section
                          class="flex flex-col flex-1 items-center text-sm sm:text-base"
                        >
                          <span
                            class="w-24 font-semibold text-center text-blue-gray-800"
                          >
                            January
                          </span>
                          <span
                            class="text-blue-gray-400"
                          >
                            2022
                          </span>
                        </section>
                        <button
                          class="flex items-center text-center rounded-full transition-colors focus:outline-none text-blue-gray-300 hover:bg-slate-100 "
                          data-testid="reach-date-right-button"
                          tabindex="3"
                          type="button"
                        >
                          <svg
                            class="m-2 "
                            fill="currentColor"
                            height="24"
                            stroke="currentColor"
                            stroke-width="0"
                            viewBox="0 0 24 24"
                            width="24"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M0 0h24v24H0V0z"
                              fill="none"
                            />
                            <path
                              d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                            />
                          </svg>
                        </button>
                      </div>
                    </label>
                    <input
                      class="absolute -left-full opacity-0"
                      data-testid="reach-date-hidden-input"
                      id="reach-date"
                      name="reach-date"
                      readonly=""
                      value="January,2022"
                    />
                  </div>
                </div>
                <div
                  class="flex flex-col rounded-lg border border-blue-gray-50"
                >
                  <p
                    class="py-6 px-8 flex justify-between items-center"
                  >
                    <span
                      class="text-lg sm:text-xl text-blue-gray-800"
                    >
                      Monthly amount
                    </span>
                    <span
                      class="block font-display font-medium text-2xl sm:text-3xl text-ellipsis overflow-hidden whitespace-nowrap text-brand-secondary"
                    >
                      $0.00
                    </span>
                  </p>
                  <div
                    class="py-6 px-8 bg-blue-gray-10"
                  >
                    <span
                      class="text-xs sm:text-sm text-blue-gray-900"
                      data-testid="monthly-amount-summary"
                    >
                      You're planning 
                      <strong
                        class="font-semibold"
                      >
                        1
                         monthly deposits 
                      </strong>
                      to reach your 
                      <strong
                        class="font-semibold"
                      >
                        $0.00
                      </strong>
                       goal by 
                      <strong
                        class="font-semibold"
                      >
                        January
                         
                        2022
                        .
                      </strong>
                    </span>
                  </div>
                </div>
                <button
                  class="w-full sm:max-w-xs !mt-8 self-center p-4 rounded-full transition-shadow bg-brand-primary text-white hover:shadow-md hover:shadow-blue-300 focus:outline-brand-secondary"
                  type="submit"
                >
                  Confirm
                </button>
              </form>
            </div>
          </div>
        </main>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="w-screen h-screen overflow-auto bg-background"
    >
      <header
        class="py-6 px-14 bg-white"
      >
        <img
          alt="Origin"
          class="h-8"
          data-testid="logo"
          src="logo.svg"
        />
      </header>
      <main
        class="mt-8 sm:mt-12 flex flex-col items-center"
      >
        <div
          class="sm:max-w-[40rem]"
        >
          <h3
            class="mb-6 text-center text-lg sm:text-xl text-brand-primary"
          >
            Let's plan your
             
            <strong
              class="font-semibold"
            >
              saving goal.
            </strong>
          </h3>
          <div
            class="pt-8 p-10 rounded-lg shadow-level4 bg-neutral-white"
          >
            <form
              class="flex flex-col flex-wrap space-y-6"
            >
              <section
                class="flex items-center space-x-4"
              >
                <img
                  alt="Buy a house"
                  src="buy-a-house.svg"
                  width="64"
                />
                <div
                  class="space-y-1"
                >
                  <h1
                    class="text-xl sm:text-2xl font-display font-medium text-blue-gray-800"
                  >
                    Buy a house
                  </h1>
                  <h2
                    class="text-sm sm:text-base text-blue-gray-400"
                  >
                    Saving Goal
                  </h2>
                </div>
              </section>
              <div
                class="flex flex-wrap gap-4"
              >
                <div
                  class="flex flex-1 flex-col"
                >
                  <label
                    class="mb-1 text-xs sm:text-sm text-blue-gray-800"
                    for="amount"
                  >
                    Total amount
                  </label>
                  <div
                    class="pl-1 p-3 group h-14 flex flex-nowrap items-center space-x-1 rounded border font-display font-medium text-xl sm:text-2xl border-blue-gray-50 text-blue-gray-600 transition-colors overflow-hidden focus-within:outline focus-within:outline-brand-secondary focus-within:outline-2 undefined"
                  >
                    <span
                      class="transition-colors text-blue-gray-100 group-focus-within:text-brand-secondary"
                    >
                      <svg
                        fill="currentColor"
                        height="24"
                        stroke="currentColor"
                        stroke-width="0"
                        viewBox="0 0 16 16"
                        width="24"
                        xmlns="http://www.w3.org/2000/svg"
                      >
                        <path
                          d="M4 10.781c.148 1.667 1.513 2.85 3.591 3.003V15h1.043v-1.216c2.27-.179 3.678-1.438 3.678-3.3 0-1.59-.947-2.51-2.956-3.028l-.722-.187V3.467c1.122.11 1.879.714 2.07 1.616h1.47c-.166-1.6-1.54-2.748-3.54-2.875V1H7.591v1.233c-1.939.23-3.27 1.472-3.27 3.156 0 1.454.966 2.483 2.661 2.917l.61.162v4.031c-1.149-.17-1.94-.8-2.131-1.718H4zm3.391-3.836c-1.043-.263-1.6-.825-1.6-1.616 0-.944.704-1.641 1.8-1.828v3.495l-.2-.05zm1.591 1.872c1.287.323 1.852.859 1.852 1.769 0 1.097-.826 1.828-2.2 1.939V8.73l.348.086z"
                        />
                      </svg>
                    </span>
                    <input
                      class="flex-1 focus:outline-none"
                      id="amount"
                      inputmode="decimal"
                      name="amount"
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  class="flex flex-1 flex-col"
                >
                  <label
                    for="reach-date"
                    tabindex="1"
                  >
                    <span
                      class="mb-1 text-xs sm:text-sm text-blue-gray-800"
                    >
                      Reach goal by
                    </span>
                    <div
                      class="h-14 py-1 px-2 flex rounded border items-center border-blue-gray-50  undefined"
                    >
                      <button
                        class="w-10 h-10 text-center rounded-full transition-colors focus:outline-none text-blue-gray-50 hover:bg-transparent cursor-not-allowed"
                        data-testid="reach-date-left-button"
                        tabindex="2"
                        type="button"
                      >
                        <svg
                          class="m-2"
                          fill="currentColor"
                          height="24"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M0 0h24v24H0V0z"
                            fill="none"
                          />
                          <path
                            d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                          />
                        </svg>
                      </button>
                      <section
                        class="flex flex-col flex-1 items-center text-sm sm:text-base"
                      >
                        <span
                          class="w-24 font-semibold text-center text-blue-gray-800"
                        >
                          January
                        </span>
                        <span
                          class="text-blue-gray-400"
                        >
                          2022
                        </span>
                      </section>
                      <button
                        class="flex items-center text-center rounded-full transition-colors focus:outline-none text-blue-gray-300 hover:bg-slate-100 "
                        data-testid="reach-date-right-button"
                        tabindex="3"
                        type="button"
                      >
                        <svg
                          class="m-2 "
                          fill="currentColor"
                          height="24"
                          stroke="currentColor"
                          stroke-width="0"
                          viewBox="0 0 24 24"
                          width="24"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M0 0h24v24H0V0z"
                            fill="none"
                          />
                          <path
                            d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                          />
                        </svg>
                      </button>
                    </div>
                  </label>
                  <input
                    class="absolute -left-full opacity-0"
                    data-testid="reach-date-hidden-input"
                    id="reach-date"
                    name="reach-date"
                    readonly=""
                    value="January,2022"
                  />
                </div>
              </div>
              <div
                class="flex flex-col rounded-lg border border-blue-gray-50"
              >
                <p
                  class="py-6 px-8 flex justify-between items-center"
                >
                  <span
                    class="text-lg sm:text-xl text-blue-gray-800"
                  >
                    Monthly amount
                  </span>
                  <span
                    class="block font-display font-medium text-2xl sm:text-3xl text-ellipsis overflow-hidden whitespace-nowrap text-brand-secondary"
                  >
                    $0.00
                  </span>
                </p>
                <div
                  class="py-6 px-8 bg-blue-gray-10"
                >
                  <span
                    class="text-xs sm:text-sm text-blue-gray-900"
                    data-testid="monthly-amount-summary"
                  >
                    You're planning 
                    <strong
                      class="font-semibold"
                    >
                      1
                       monthly deposits 
                    </strong>
                    to reach your 
                    <strong
                      class="font-semibold"
                    >
                      $0.00
                    </strong>
                     goal by 
                    <strong
                      class="font-semibold"
                    >
                      January
                       
                      2022
                      .
                    </strong>
                  </span>
                </div>
              </div>
              <button
                class="w-full sm:max-w-xs !mt-8 self-center p-4 rounded-full transition-shadow bg-brand-primary text-white hover:shadow-md hover:shadow-blue-300 focus:outline-brand-secondary"
                type="submit"
              >
                Confirm
              </button>
            </form>
          </div>
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
